src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -1376,8 +1365,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             ' // The C++ compiler refuses to inline IsIdentifierCharacter, so the
             ' // < 128 test is inline here. (This loop gets a *lot* of traffic.)
             ' TODO: make sure we get good perf here
-            While CanGet(len)
-                ch = Peek(len)
+ 
+            While Peep(len, ch)
```

Comment:
I have a version of this `while` loop that becomes.
```vb
' // The C++ compiler refuses to inline IsIdentifierCharacter, so the
' // < 128 test is inline here. (This loop gets a *lot* of traffic.)
' TODO: make sure we get good perf here
While Peep(len, ch) AndAlso (IsNarrowIdentifierCharacter(Convert.ToUInt16(ch)) OrElse IsWideIdentifierCharacter(ch))
  len += 1
End While
```
by push one of the condition into the two called functions.
```
        Friend Shared Function IsNarrowIdentifierCharacter(c As UInt16) As Boolean
            Return c < 128US AndAlso s_isIDChar(c)
        End Function

        ''' <summary>
        ''' Determines if the Unicode character can be a part of a Visual Basic identifier.
        ''' </summary>
        ''' <param name="c">The Unicode character.</param>
        ''' <returns>A boolean value set to True if character can be part of a valid identifier.</returns>
        Public Shared Function IsIdentifierPartCharacter(c As Char) As Boolean
            If c < ChrW(128) Then
                Return IsNarrowIdentifierCharacter(Convert.ToUInt16(c))
            End If

            Return IsWideIdentifierCharacter(c)
        End Function
```